drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -571,6 +585,15 @@ local function initialize_switch(driver, device)
     device:set_field(COMPONENT_TO_ENDPOINT_MAP_BUTTON, component_map, {persist = true})
   end
 
+  -- If there is a custom static profile for the device, configure buttons if needed and
+  -- then return to prevent profile from being updated.
+  if check_fingerprint_profile_overrides(device) then
+    if #button_eps > 0 then
+      configure_buttons(device)
+    end
+    return
```

Comment:
could we make this an if/else block instead? For readability, I would recommend avoiding early returns in large functions like this if they are not at the top of the function.

Reply:
I had it this way so that it would configure the buttons and then return right afterwards, to skip the profile selection logic that follows.

Reply:
To clarify, I meant could we put this entire override profile section be in an `if` statement, and then the `else` case will be the part that handles the logic in the rest of the function?
```
if profile_overide then
    config buttons and skip profile selection
else
    do the rest of the function, including profile selection
end
```

Reply:
I agree with Cooper here- this may be a good time to refactor this function a little bit.

Reply:
As I mentioned in another comment, I modified `is_supported_combination_button_switch_device_type` to include the plug device type, and also to return the device type ID so that it can be used in `initialize_switch` to set the correct profile. I think this simplifies the logic quite a bit. Could you take another look and see if it looks alright?